api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -836,6 +859,90 @@ func (s *SolutionManager) Reconcil() []error {
 	return nil
 }
 
+func (s *SolutionManager) InitCancelMap() {
+	s.jobList = make(map[string]map[string]map[string]context.CancelFunc)
+}
+
+func (s *SolutionManager) HandleCancelableJobEvent(ctx context.Context, namespace string, instance string, jobID string, isRemove string) {
```

Comment:
HandleCancelableJobEvent is not very good name to me. In fact, the real logic seems to be,

If it is delete case, we need to cancel previous jobs.
For other cases, we need some initialization to init the map. (Why can't we move the init when adding or deleting the cancel function)?